allennlp/nn/util.py

```diff
@@ -1548,7 +1548,6 @@ def add_sentence_boundary_token_ids(
         The new mask for the tensor, taking into account the appended tokens
         marking the beginning and end of the sentence.
     """
-    # TODO: matthewp, profile this transfer
```

Comment:
I benchmarked this. This function is actually faster this way than with keeping `sequence_lengths` on GPU with:

```
sequence_lengths = mask.sum(dim=1).detach()
```